nx/lib/nx/defn/expr.ex

```diff
@@ -584,6 +584,13 @@ defmodule Nx.Defn.Expr do
     tensor = to_expr(tensor)
 
     if c = maybe_constant(tensor) do
+      c =
+        if is_float(c) and Nx.Type.integer?(out.type) do
+          floor(c)
```

Comment:
what the binary backend uses is:

```suggestion
          trunc(c)
```